backend/FileSystem/FileManager.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\FileSystem;
+
+use Exception;
+
+class FileManager
+{
+    public function __construct(
+        protected DirectoryManager &$directoryManager
+    ) {
+    }
+
+    public function create(string $relativePath, string $filename = "\0", mixed $content = ""): void
+    {
+        $path = $this->directoryManager->getApiPath($relativePath);
+
+        if ($filename !== "\0") {
+            file_put_contents(
+                filename: $path . $filename,
+                data: $content
+            );
+        }
+    }
+
+    public function copyResource(string $relativeOrigin, string $relativeDestination, $filename, $newName = "\0"): void
+    {
+        if ($newName === "\0") {
+            $newName = $filename;
+        }
+        $destination = $this->directoryManager->getApiPath($relativeDestination);
+        $origin = $this->directoryManager->getResourcePath($relativeOrigin);
+        if (!copy($origin . $filename, $destination . $newName)) {
+            throw new Exception("Couldn't copy. File " . $origin . $filename . " not found.");
```

Comment:
I'm not frocing you to do this, but in PHP you can create your own exceptions to be more precise.

Reply:
I was considering custom exceptions when I research them, but I quickly threw it together to test feature I was working on first, then I kinda forgot about them. I'll toy around with this idea.

Reply:
We can also create a new task to split exceptions. :)

Reply:
I think I'll open separate issue for that actually, it's not something that needs to be done ASAP, but would be nice to implement.